src/operator/controllers/secrets/manager_test.go

```diff
@@ -511,6 +509,18 @@ func (s *ManagerSuite) TestManager_RefreshTLSSecrets_RefreshNeeded() {
 		},
 	).Return(nil)
 
+	pod := corev1.Pod{}
```

Comment:
Add a test that checks that it modifies the owner object, and that it _doesn't_ modify it when you remove the annotation from the pod.